Packs/SophosCentral/Integrations/SophosCentral/SophosCentral.py

```diff
@@ -0,0 +1,1764 @@
+        outputs = {}
+    return CommandResults(raw_response=result, readable_output=readable_output, outputs=outputs,
+                          outputs_prefix='SophosCentral.DeletedExploitMitigation')
+
+
+def create_detected_exploit_output(item: Dict) -> Dict:
+    """
+    Create the complete output dictionary for a detected exploit.
+
+    Args:
+        item (dict(str)): A source dictionary from the API response.
+
+    Returns:
+        object_data (dict(str)): The output dictionary.
+    """
+    data_fields = ['id', 'thumbprint', 'count', 'description', 'firstSeenAt', 'lastSeenAt']
+    outputs = {field: item.get('field') for field in data_fields}
+
+    last_user = item.get('lastUser')
+    if last_user:
+        outputs['lastUserName'] = last_user.get('name')
+        outputs['lastUserId'] = last_user.get('id')
+
+    last_endpoint = item.get('lastEndpoint')
+    if last_endpoint:
+        outputs['lastEndpointHostname'] = last_endpoint.get('hostname')
+        outputs['lastEndpointId'] = last_endpoint.get('id')
+
+    return outputs
+
+
+def sophos_central_detected_exploit_list_command(client: Client, args: dict) -> CommandResults:
+    """
+    List all detected exploits.
+
+    Args:
+        client (Client): Sophos Central API client.
+        args (dict): All command arguments
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    results = client.list_detected_exploit(min(int(args.get('page_size', '')), 100),
+                                           args.get('page'), args.get('thumbprint_not_in'))
+    items = results.get('items')
+    table_headers = ['id', 'description', 'thumbprint', 'count', 'firstSeenAt', 'lastSeenAt']
+    outputs = []
+    if items:
+        for item in items:
+            current_object_data = create_detected_exploit_output(item)
+            outputs.append(current_object_data)
+    readable_output = f'### Current page: {int(args.get("page", ""))}\n'
+    readable_output += tableToMarkdown(name=f'Listed {len(outputs)} Detected Exploits:', t=outputs,
+                                       headers=table_headers, removeNull=True)
+    return CommandResults(outputs_key_field='id', outputs_prefix='SophosCentral.DetectedExploit',
+                          raw_response=results, outputs=outputs, readable_output=readable_output)
+
+
+def sophos_central_detected_exploit_get_command(client: Client, args: dict) -> CommandResults:
+    """
+    Get a single detected exploit.
+
+    Args:
+        client (Client): Sophos Central API client.
+        args (dict): All command arguments
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    result = client.get_detected_exploit(args.get('detected_exploit_id', ''))
+    table_headers = ['id', 'description', 'thumbprint', 'count', 'firstSeenAt', 'lastSeenAt']
+    object_data = {}
+    if result:
+        object_data = create_detected_exploit_output(result)
```

Comment:
if no result?
probably something like:
readable_output = f'Exploit {detected_exploit_id} was not detected.'

please try it and make sure that the API does not return an error, if he does - catch it to print a normal war room entry.

Reply:
The API will return an error in case it isn't found, so I'll wrap this in a try / catch statement.

Reply:
there is a try/catch in the main, so make sure it is needed before you add it. i'm not certain this is the case - but your call.

Reply:
The one in main will still return an error to the war room. For commands like get / update I think it is a better idea to print a more friendly entry if a certain object was not found.